### ISSUE TITLE
My USB hard disk is recognized as a local hard disk.

### ISSUE THREAD

**Issue by shanzigang** [fbelavenuto/arpl#304]
My USB hard disk is recognized as a local hard disk.

DSM: DS918+ 7.1.1-42962 Update 2
motherboard: Gigabyte Z370
CPU：G5500T
SATA controllers:
Intel controller on motherboard: 4 SATA ports
9211 SAS IT 8 Ports*2
Hard disk num:16*STAT  2*SSD

Whether I take the default configuration or the following configuration，My mobile hard disk is recognized as a local hard disk.
esataportcfg="0xff000000"
internalportcfg="0xffffff"
maxdisks="32"

**Comment by fbelavenuto** [comment]
Please do a test: Remove manual settings from 'esataportcfg', 'internalportcfg' and 'maxdisks', boot without USB HD and plug in after DSM starts.

**Comment by shanzigang** [comment]
> 请做一个测试：从 'esataportcfg'、'internalportcfg' 和 'maxdisks' 中删除手动设置，在没有 USB HD 的情况下启动并在 DSM 启动后插入。

Thank you very much for your reply. According to the method you provided (deleting 'esataportcfg', 'internalportcfg' and 'maxdisks' in the configuration file), the external hard disk cannot be displayed normally. 

I tried a new set of parameters and the problem is now solved .

esataportcfg="0x3ff00000"
usbportcfg="0xfc0000000"
internalportcfg="0xfffff"
maxdisk="20"

111111000000000000000000000000000000   usb
000000111111111100000000000000000000   esata
000000000000000011111111111111111111   sata


My understanding of the maxdisk parameter is wrong. I originally thought that the maxdisk parameter includes the esata interface. In fact, this parameter only refers to the internal disk. Now I set this parameter to the actual number of interfaces on my machine (SATA+SAS=20), and it's back to normal.
